src/main/java/at/hannibal2/skyhanni/features/mining/glacitemineshaft/CorpseLocator.kt

```diff
@@ -0,0 +1,141 @@
+package at.hannibal2.skyhanni.features.mining.glacitemineshaft
+
+import at.hannibal2.skyhanni.SkyHanniMod
+import at.hannibal2.skyhanni.data.IslandType
+import at.hannibal2.skyhanni.data.PartyAPI
+import at.hannibal2.skyhanni.events.LorenzChatEvent
+import at.hannibal2.skyhanni.events.LorenzRenderWorldEvent
+import at.hannibal2.skyhanni.events.LorenzWorldChangeEvent
+import at.hannibal2.skyhanni.events.SecondPassedEvent
+import at.hannibal2.skyhanni.test.GriffinUtils.drawWaypointFilled
+import at.hannibal2.skyhanni.utils.ChatUtils
+import at.hannibal2.skyhanni.utils.EntityUtils
+import at.hannibal2.skyhanni.utils.HypixelCommands
+import at.hannibal2.skyhanni.utils.LocationUtils.canBeSeen
+import at.hannibal2.skyhanni.utils.LocationUtils.distanceToPlayer
+import at.hannibal2.skyhanni.utils.LorenzUtils
+import at.hannibal2.skyhanni.utils.LorenzUtils.isInIsland
+import at.hannibal2.skyhanni.utils.LorenzVec
+import at.hannibal2.skyhanni.utils.RenderUtils.drawDynamicText
+import at.hannibal2.skyhanni.utils.StringUtils.cleanPlayerName
+import at.hannibal2.skyhanni.utils.RegexUtils.matchMatcher
+import at.hannibal2.skyhanni.utils.getLorenzVec
+import at.hannibal2.skyhanni.utils.repopatterns.RepoPattern
+import net.minecraft.client.Minecraft
+import net.minecraft.entity.item.EntityArmorStand
+import net.minecraft.util.StringUtils
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+
+// TODO: Maybe implement automatic warp-in for chosen players if the user is not in a party.
+object CorpseLocator {
+    private val config get() = SkyHanniMod.feature.mining.mineshaftWaypoints.corpseLocator
+
+    /**
+     * I have no clue what to do for the regex anymore, I'm slowly losing it
+     * REGEX-TEST: §9Party §8> §b[MVP§0+]§b nobaboy§f: §rx: -164, y: 8, z: -154 | (Lapis Corpse)
+     * REGEX-TEST: Guild > §r§6♣ §b[MVP§0+§b] nobaboy§f: x: 141, y: 14, z: -131
+     * REGEX-TEST: §8[§r§6407§r§8] §r§6♣ §b[MVP§0+§b] nobaboy§f: x: -9, y: 135, z: 20 | (Tungsten Corpse)
+     */
+    private val corpseCoordsWaypoint by RepoPattern.pattern(
+        "corpse.waypoints",
+        "^(?:.+?[^x:y0-9] )?(?<playerName>.+)§f: (?:§r)?x: (?<x>-?\\d+),? y: (?<y>-?\\d+),? z: (?<z>-?\\d+)(?:.+)?"
+    )
+
+    private val parsedLocations: MutableList<LorenzVec> = mutableListOf()
+
+    private fun findCorpse() {
+        Minecraft.getMinecraft().theWorld ?: return
+
+        EntityUtils.getAllEntities().filterIsInstance<EntityArmorStand>()
+            .filterNot { corpse ->
+                MineshaftWaypoints.waypoints.any { it.location.distance(corpse.getLorenzVec()) <= 3 }
+            }
+            .filter { entity ->
+                entity.showArms && entity.hasNoBasePlate() && !entity.isInvisible
+            }
+            .forEach { entity ->
+                val helmetDisplayName = StringUtils.stripControlCodes(entity.getCurrentArmor(3).displayName)
```

Comment:
use neu internal name instead of item name

Reply:
how do I modify MineshaftWaypointType to work with internal name

Reply:
something along the lines of
entity.getCurrentArmor(3).getInternalName()

and then you store the neu internal names in the enum instead, also id make a function in the enums companion object  like getByHelmentOrNull. You could find similar functions to these in other enums such as CropType

Reply:
so keep helmet name, add internal id as a val to the enum, check with internal id, if it fails check with helmet name? or at least that's the gist I'm getting here

Reply:
use internal name instead of item name